File: examples-sdk/kotlin/606-db-vilorm-ecommerce/main.kt
// 606-db-vilorm-ecommerce — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 606-db-vilorm-ecommerce/main.kt --release

fun main() {
    val server = VilServer("vilorm-ecommerce", 8086)
    val shop = ServiceProcess("shop")
    shop.endpoint("POST", "/products", "create_product")
    shop.endpoint("GET", "/products", "list_products")
    shop.endpoint("GET", "/products/:id", "get_product")
    shop.endpoint("POST", "/orders", "create_order")
    shop.endpoint("GET", "/orders", "list_orders")
    shop.endpoint("GET", "/orders/:id/total", "order_total")
    shop.endpoint("DELETE", "/orders/:id", "cancel_order")
    server.service(shop)
    server.compile()
}

File: examples-sdk/swift/606-db-vilorm-ecommerce/main.swift
// 606-db-vilorm-ecommerce — Swift SDK equivalent
// Compile: vil compile --from swift --input 606-db-vilorm-ecommerce/main.swift --release

let server = VilServer(name: "vilorm-ecommerce", port: 8086)
let shop = ServiceProcess(name: "shop")
shop.endpoint(method: "POST", path: "/products", handler: "create_product")
shop.endpoint(method: "GET", path: "/products", handler: "list_products")
shop.endpoint(method: "GET", path: "/products/:id", handler: "get_product")
shop.endpoint(method: "POST", path: "/orders", handler: "create_order")
shop.endpoint(method: "GET", path: "/orders", handler: "list_orders")
shop.endpoint(method: "GET", path: "/orders/:id/total", handler: "order_total")
shop.endpoint(method: "DELETE", path: "/orders/:id", handler: "cancel_order")
server.service(shop)
server.compile()
